src/backend/src/services/tasks.services.ts

```diff
@@ -51,7 +55,18 @@ export default class TasksService {
     const isProjectLeadOrManager =
       createdBy.userId === requestedWbsElement.projectLeadId || createdBy.userId === requestedWbsElement.projectManagerId;
 
-    if (!isLeadership(createdBy.role) && !isProjectLeadOrManager && !teams.some((team) => isUserOnTeam(team, createdBy))) {
+    const curWorkPackages = requestedWbsElement.project?.workPackages;
+
+    const isWorkPackageLead = curWorkPackages?.some((workPackage) => {
+      return workPackage.wbsElement.projectLeadId === createdBy.userId;
+    });
+
+    if (
+      !isLeadership(createdBy.role) &&
+      !isProjectLeadOrManager &&
+      isWorkPackageLead &&
+      !teams.some((team) => isUserOnTeam(team, createdBy))
+    ) {
       throw new AccessDeniedException(
         'Only admins, app-admins, project leads, project managers, or current team users can create tasks'
```

Comment:
We want to update this error message, where we also wanna include that work package leads under a project can also create tasks.